documents-webapp/src/main/webapp/vue-app/documents/components/body/actions/CopyLinkMenuAction.vue

```diff
@@ -32,7 +32,7 @@ export default {
       if (this.file.folder){
         path = `${path}?folderId=${this.file.id}`;
       } else {
-        path = `${path}?documentPreviewId=${this.file.id}`;
+        path = `${path}?documentPreviewId=${this.file.sourceID ? this.file.sourceID:this.file.id}`;
```

Comment:
@ahamdi i add the same condition to the copied link to get the original file and that fix the problem .WDYT ?